kernel/packages/unity-interface/loader.ts

```diff
@@ -0,0 +1,105 @@
+import future from 'fp-future'
+import { parseUrn } from '@dcl/urn-resolver'
+
+declare const globalThis: any
+
+export type DclRenderer = {}
+export type UnityLoader = {}
+export type LoadRendererResult = {
+  DclRenderer: DclRenderer
+  UnityLoader: UnityLoader
+  buildConfigPath: string
+}
+
+/**
+ * This function is used by preview, instead of using "." as root,
+ * preview uses '/@/artifacts'
+ */
+function getRendererArtifactsRoot() {
+  if (typeof globalThis.RENDERER_ARTIFACTS_ROOT == 'undefined') {
+    throw new Error('RENDERER_ARTIFACTS_ROOT is undefined')
+  } else {
+    return new URL(globalThis.RENDERER_ARTIFACTS_ROOT, document.location.toString()).toString()
+  }
+}
+
+// TODO: return type DclRenderer
+async function injectRenderer(baseUrl: string): Promise<LoadRendererResult> {
+  const scriptUrl = new URL('index.js', baseUrl).toString()
+  const buildConfigPath = new URL('unity.json', baseUrl).toString()
+  await injectScript(scriptUrl)
+
+  if (typeof globalThis.UnityLoader == 'undefined') {
+    throw new Error('Error while loading UnityLoader from ' + scriptUrl)
+  }
+
+  if (typeof globalThis.DclRenderer == 'undefined') {
+    throw new Error('Error while loading the renderer from ' + scriptUrl)
+  }
+
+  return {
+    DclRenderer: globalThis.DclRenderer,
+    UnityLoader: globalThis.UnityLoader,
+    buildConfigPath
+  }
+}
+// TODO: return type DclRenderer
+async function loadDefaultRenderer(): Promise<LoadRendererResult> {
+  // TODO: after we remove unity from this folder, remove the following
+  // block and uncomment the block next to it
+  {
+    const scriptUrl = new URL('DCLUnityLoader.js', getRendererArtifactsRoot()).toString()
```

Comment:
This will prompt an error in Builder since we are using another loader
(https://github.com/decentraland/builder/blob/master/public/UnityLoader.js)
Otherwise these changes are working fine!

Reply:
do you know what changes does that unityloader contains from the original one? or the reason behind that local copy?

Reply:
I have no idea. I could check.
I remember trying once, a lot time ago, to change it to use` DCLUnityLoader.js` instead but I had no success

Reply:
@menduz So it seems the only important difference between the two loaders is here:
https://github.com/decentraland/explorer/blob/master/kernel/static/unity/Build/DCLUnityLoader.js#L5199-L5200

For builder to work we would need to replace these two lines:
`document.getElementById('gameContainer').classList.add('loaded')`
`document.querySelector('.progress').classList.add('loaded')`
For this one:
`currentProgress = 100 * t + '%'`

Reply:
I verified, it is not being used anywhere else

This will suffice to make the builder work with the new release, after we merge to master we can create a PR in builder to test `@next`:

```ts
// inside builder bundle
UnityLoader.Progress.handler = function(){}
```